test/e2e/runner/start.go

```diff
@@ -66,8 +73,26 @@ func Start(testnet *e2e.Testnet) error {
 		}
 	}
 
-	// Start up remaining nodes
 	for _, node := range nodeQueue {
+		if node.StartAt > networkHeight {
+			// if we're starting a node that's ahead of
+			// the last known height of the network, then
+			// we should make sure that the rest of the
+			// network has reached at least the height
+			// that this node will start at before we
+			// start the node.
+
+			networkHeight = node.StartAt
+
+			logger.Info("Waiting for network to advance before starting catch up node",
+				"node", node.Name,
+				"height", networkHeight)
+
+			if _, _, err := waitForHeight(testnet, networkHeight); err != nil {
```

Comment:
It looks like this change alters the code to only call `waitForHeight` if `node.StartAt > networkHeight`. I think `waitForHeight` returns immediately if `networkHeight <= node.StartAt` and will wait if `node.StartAt < networkHeight` so I'm a little confused as to what this change is doing. Mind giving me some guidance?

Reply:
I think I keep stumbling over the best name for the `networkHeight` varabile. It's the "max last known network height." and the idea is that we set `networkHeight` to a new (higher) value right before (in all cases) we wait for the new height, we can expect that the value of `networkHeight` is *at least* the height, and the check here is just "make sure we're at least this high" before starting a node at the given height.